subsys/bluetooth/services/ias.c

```diff
@@ -0,0 +1,122 @@
+/** @file
+ *  @brief IAS Service sample
+ */
+
+/*
+ * Copyright (c) 2022 Codecoup
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "bluetooth/att.h"
+#include "net/buf.h"
+#include "bluetooth/services/ias.h"
+#include <init.h>
+#include <stdint.h>
+#include <zephyr.h>
+#include <logging/log.h>
+#include <bluetooth/gatt.h>
+
+#define LOG_LEVEL CONFIG_BT_IAS_LOG_LEVEL
+LOG_MODULE_REGISTER(ias);
+
+#define BT_IAS_ALERT_MAX_LEN 1
+
+typedef enum __packed {
+	BT_IAS_ALERT_LVL_NO_ALERT,
+	BT_IAS_ALERT_LVL_MILD_ALERT,
+	BT_IAS_ALERT_LVL_HIGH_ALERT,
+} bt_ias_alert_lvls;
+
+#define GATT_PERM_WRITE_MASK    (BT_GATT_PERM_WRITE | \
+				 BT_GATT_PERM_WRITE_ENCRYPT | \
+				 BT_GATT_PERM_WRITE_AUTHEN)
+
+#if defined(CONFIG_BT_IAS_DEFAULT_PERM_W_AUTHEN)
+#define IAS_GATT_PERM_DEFAULT BT_GATT_PERM_WRITE_AUTHEN
+#elif defined(CONFIG_BT_IAS_DEFAULT_PERM_W_ENCRYPT)
+#define IAS_GATT_PERM_DEFUALT CONFIG_BT_IAS_DEFAULT_PERM_W_ENCRYPT 
+#else
+#define IAS_GATT_PERM_DEFAULT BT_GATT_PERM_WRITE
+#endif
+
+bt_ias_alert_lvls g_alert_level;
+
+static void alert_level(bt_ias_alert_lvls alert_lvl)
+{
+	if (alert_lvl == g_alert_level) {
+		return;
+	}
+
+	if (alert_lvl == BT_IAS_ALERT_LVL_NO_ALERT) {
+		STRUCT_SECTION_FOREACH(bt_ias_cb, cb)
+		{
+			if (cb->stop_alert) {
+				cb->stop_alert();
+				g_alert_level = BT_IAS_ALERT_LVL_NO_ALERT;
+				LOG_DBG("Alert stop");
+			}
+		}
+	} else if (alert_lvl == BT_IAS_ALERT_LVL_MILD_ALERT) {
+		STRUCT_SECTION_FOREACH(bt_ias_cb, cb)
+		{
+			if (cb->start_alert) {
+				cb->start_alert();
+				g_alert_level = BT_IAS_ALERT_LVL_MILD_ALERT;
+				LOG_DBG("Mild alert start");
+			}
+		}
+	} else if (alert_lvl == BT_IAS_ALERT_LVL_HIGH_ALERT) {
+		STRUCT_SECTION_FOREACH(bt_ias_cb, cb)
+		{
+			if (cb->start_alert_high) {
+				cb->start_alert_high();
+				g_alert_level = BT_IAS_ALERT_LVL_HIGH_ALERT;
+				LOG_DBG("High alert start");
+			}
+		}
+	}
+}
+
+static void disconnected(struct bt_conn *conn, uint8_t reason)
+{
+	if (g_alert_level) {
+		alert_level(BT_IAS_ALERT_LVL_NO_ALERT);
+	}
```

Comment:
Given that you also have the check in `alert_level` it could possibly be omitted, but otherwise a more clear check should be done:
```suggestion
	if (g_alert_level == BT_IAS_ALERT_LVL_NO_ALERT) {
		alert_level(BT_IAS_ALERT_LVL_NO_ALERT);
	}
```